salt/addons/deploy_tiller.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+set -e
+
+KUBECTL_PATH="${KUBECTL_PATH:-/usr/bin/kubectl}"
+
+# dir where manifests are
+DIR=/etc/kubernetes/addons
+
+############################################################################
+
+abort() { echo "FATAL: $@" ; exit 1 ; }
+
+NS_ARGS="--namespace=kube-system"
+SRV_ARGS="--server=http://127.0.0.1:8080"
+
+echo "Deploying tiller on Kubernetes"
+
+TILLER=`eval "$KUBECTL_PATH $NS_ARGS $SRV_ARGS get deployment | grep tiller | cat"`
+if [ ! "$TILLER" ]; then
+	# use kubectl to create tiller deployment
+	$KUBECTL_PATH $NS_ARGS $SRV_ARGS create -f "$DIR/tiller.yaml"
```

Comment:
Another option to consider is to combine the kubernetes deployment + service for tiller into a single app manifest (yaml). This way the add-on can be deployed by running a single "kubectl apply -f".